### PR TITLE
Add port Number

### PR DESCRIPTION
**Pull request by ThulasikaV**

### CODE REVIEW

File: jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml [hunk 1]
```diff
@@ -19,18 +19,20 @@
 <!-- Method Documentation URL: https://wiki.jenkins-ci.org/display/JENKINS/Remote+access+API -->
 <template name="init" xmlns="http://ws.apache.org/ns/synapse">
    <parameter name="apiUrl" description="Base endpoint URL of Jenkins API." />
+   <parameter name="port" description="Port number" />
```
Comment (Shakila) [inline jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml]: Add more descriptive description.

File: jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml [hunk 2]
```diff
@@ -19,18 +19,20 @@
...
    <sequence>
       <property name="uri.var.apiUrl" expression="$func:apiUrl" />
       <property name="uri.var.username" expression="$func:username" />
       <property name="uri.var.password" expression="$func:password" />
-      <filter source="boolean(get-property('uri.var.apiUrl'))" regex="true">
+      <property name="uri.var.port" expression="$func:port" />
+      <filter source="boolean(get-property('uri.var.apiUrl') and get-property('uri.var.port'))" regex="true">
          <then>
-         	<property name="uri.var.apiUrl" expression="fn:concat(get-property('uri.var.apiUrl'), ':8080')"/>
+            <property name="uri.var.apiUrl" expression="fn:concat(get-property('uri.var.apiUrl'), ':',get-property('uri.var.port'))"/>
```
Comment (Shakila) [inline jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml]: Format this line

File: jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/test/java/org/wso2/carbon/connector/integration/test/jenkins/JenkinsConnectorIntegrationTest.java [hunk 3]
```diff
@@ -60,8 +60,7 @@ public void setEnvironment() throws Exception {
         apiRequestHeadersMap.put("Authorization", "Basic " + base64AuthString);
         
         apiRequestHeadersMap.putAll(esbRequestHeadersMap);
-        apiEndpointUrl = connectorProperties.getProperty("apiUrl") + ":8080";
-        
+        apiEndpointUrl = connectorProperties.getProperty("apiUrl") + ":"+connectorProperties.getProperty("port");
```
Comment (Shakila) [inline jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/test/java/org/wso2/carbon/connector/integration/test/jenkins/JenkinsConnectorIntegrationTest.java]: Need to format this line

File: jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/proxies/jenkins/jenkins.xml [hunk 4]
```diff
@@ -36,14 +37,16 @@
...
+                   <port>{$ctx:port}</port>
+                   <password>{$url:password}</password>
                </jenkins.init>
             </then>
             <else>
                <jenkins.init>
                   <username>{$ctx:username}</username>
                   <apiUrl>{$ctx:apiUrl}</apiUrl>
-                  <password>{$ctx:password}</password>
+                   <port>{$ctx:port}</port>
```
Comment (Shakila) [inline jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/proxies/jenkins/jenkins.xml]: Please format all changed code lines in this PR.

File: jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml [hunk 5]
```diff
@@ -19,18 +19,20 @@
...
+   <parameter name="port" description="Port number" />
    <parameter name="username" description="Username of the Jenkins account being accessed." />
    <parameter name="password" description="Password corresponding to the username provided for the Jenkins account." />
    <sequence>
       <property name="uri.var.apiUrl" expression="$func:apiUrl" />
       <property name="uri.var.username" expression="$func:username" />
       <property name="uri.var.password" expression="$func:password" />
-      <filter source="boolean(get-property('uri.var.apiUrl'))" regex="true">
+      <property name="uri.var.port" expression="$func:port" />
+      <filter source="boolean(get-property('uri.var.apiUrl') and get-property('uri.var.port'))" regex="true">
```
Comment (malakasilva) [inline jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml]: ctx.property_name is preferred over get-property method.

Comment (malakasilva) [inline jenkins/jenkins-connector/jenkins-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/jenkins-config/init.xml]: We need to make port as an optional param.

**Comment by ThulasikaV** [comment]
I did the required changes.

**Pull request merged**